packages/shopify-codemod/transforms/global-assignment-to-default-export.js

```diff
@@ -0,0 +1,47 @@
+export default function globalAssignmentToDefaultExport({source}, {jscodeshift: j}, {printOptions = {}, appGlobalIdentifiers}) {
+  function findFirstMember(memberExpression) {
+    if (j.MemberExpression.check(memberExpression)) {
+      return findFirstMember(memberExpression.object);
+    }
+    return memberExpression;
+  }
+
+  return j(source)
+    .find(j.Program)
+    .forEach((programPath) => {
+      let expose = null;
+      j(programPath)
+        .find(j.ExpressionStatement, {
+          expression: {
+            type: 'AssignmentExpression',
+            operator: '=',
+            left: {
+              type: 'MemberExpression',
+              object: (object) => appGlobalIdentifiers.indexOf(findFirstMember(object).name) >= 0,
+            },
+          },
+        })
+        .replaceWith((path) => {
+          const {
+            node: {
+              expression: {
+                left: member,
+                right: statement,
+              },
+            },
+          } = path;
+
+          if (expose == null) {
+            expose = j(member).toSource();
+          } else {
+            throw new Error('Found multiple exports in a single file, please break up the file first');
+          }
+          return j.exportDefaultDeclaration(statement);
+        });
+      if (expose !== null) {
```

Comment:
I'm not a huge fan of that rule since it adds complexity to the recommendation for questionable benefit. Can we just stick to the one? Either way, should probably just be `!= null` to match above.